simple_salesforce/api.py

```diff
@@ -124,7 +126,8 @@ def __init__(
             self.auth_type = "password"
 
             # Pass along the username/password to our login helper
-            self.session_id, self.sf_instance = SalesforceLogin(
+            self._salesforce_login_partial = partial(
```

Comment:
I'm basically using `partial` to cache the values passed to `SalesforceLogin` in `_salesforce_login_partial` so later when our session expires, we can just call `SalesforceLogin` again and get a new token.